docs/opensafely-cli.md

```diff
@@ -117,15 +117,75 @@ See the [Codelist](codelist-intro.md) section for more information on codelists.
 
 
 To run your code on your machine, the `opensafely` tool uses the same Docker
-images that run in the secure server environments. These are updated
-periodically, for example when new libraries are installed. If you have error
-messages about missing libraries, your Docker images may need upgrading.
-To pull the most recent Docker images to your machine, run:
+images that run in the secure server environments. There is the
+`cohortextractor` image, for processing study definitions, and then `r`,
+`stata-mp`, and `python` images, for running your analysis code. These last
+three provide a pre-built environment for their specific tool, with a set of libraries.
+
+
+These are updated periodically, for example when new libraries are installed.
+If you have error messages about missing libraries, your Docker images may need
+upgrading.  To pull the most recent Docker images to your machine, run:
 
 ```bash
 opensafely pull
 ```
 
+
+### Developing code with the OpenSAFELY images.
+
+The R and Stata tools are often used interactively to develop and test analysis code, and python users often use `ipython` or `jupyterlab` to develop and test code.
+
+The `opensafely exec` command provides a simple way to do this, by running interactive tools using the Docker images provided by OpenSAFELY. This can help ensure that your code works correctly in OpenSAFELY as you develop it, rather than accidentally relying on tools and libraries installed on your own machine.
```

Comment:
Per my previous point - perhaps we want to be encouraging a pipeline-first approach, in which case it would be worth adding something like "The pipeline automatically ensures the languages and tools used on your own machine match those in OpenSAFELY, but if you're debugging problems outside the pipeline...."